honestcash-v2/src/app/shared/components/avatar/avatar.component.ts

```diff
@@ -0,0 +1,32 @@
+import {Component, Input} from '@angular/core';
+
+@Component({
+             selector: 'app-avatar',
+             template: `<img class="avatar" [ngClass]="{'large': large, 'small': !large}" [src]="src"/>`,
+             styles: [
+               '.avatar {\n' +
+               '  border-style: solid;\n' +
+               '  border-color: #e7e7e7;\n' +
+               '  border-width: 1px;\n' +
+               '  border-radius: 50%;\n' +
+               '}' +
+               '.avatar.large {\n' +
+               '  height: 45px;\n' +
+               '  width: 45px;\n' +
+               '}' +
+               '.avatar.small {\n' +
+               '  height: 34px;\n' +
+               '  width: 34px;\n' +
+               '}'
+             ]
+           })
+export class AvatarComponent {
+  private _src = '';
+  @Input() large = false;
+  @Input()
+  set src(src: string) {
```

Comment:
use explicit setSource  / getSource. It is easier to be more explicit, expecially when we are in the component logic.

Reply:
these are not public functions, you will not use them anywhere else other than the component. the component provides readable code already

Reply:
additionally: you cannot run a function to do what angular does by default (mimicing @Input decorator)